cli/lsp/handlers.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+use super::lsp_extensions;
+use super::state::ServerState;
+use super::state::ServerStateSnapshot;
+use super::text;
+use super::tsc;
+use super::utils;
+
+use deno_core::error::custom_error;
+use deno_core::error::AnyError;
+use deno_core::serde_json;
+use deno_core::ModuleSpecifier;
+use dprint_plugin_typescript as dprint;
+use lsp_types::DocumentFormattingParams;
+use lsp_types::DocumentHighlight;
+use lsp_types::DocumentHighlightParams;
+use lsp_types::Hover;
+use lsp_types::HoverParams;
+use lsp_types::Location;
+use lsp_types::ReferenceParams;
+use lsp_types::TextEdit;
+use std::path::PathBuf;
+
+pub fn handle_formatting(
+  state: ServerStateSnapshot,
+  params: DocumentFormattingParams,
+) -> Result<Option<Vec<TextEdit>>, AnyError> {
+  let specifier = utils::normalize_url(params.text_document.uri.clone());
+  let file_cache = state.file_cache.read().unwrap();
+  let file_id = file_cache.lookup(&specifier).unwrap();
+  let file_text = file_cache.get_contents(file_id)?;
+
+  let file_path = if let Ok(file_path) = params.text_document.uri.to_file_path()
+  {
+    file_path
+  } else {
+    PathBuf::from(params.text_document.uri.path())
+  };
+  let config = dprint::configuration::ConfigurationBuilder::new()
+    .deno()
+    .build();
+
+  let new_text = dprint::format_text(&file_path, &file_text, &config)
+    .map_err(|e| custom_error("FormatError", e))?;
```

Comment:
Maybe abstract into a helper function in `tools/fmt.rs`? I'm a bit worried these calls might get out of sync with `tools/fmt.rs` (probably unlikely)

Reply:
I wanted to avoid/limit changes outside of lsp at this point...  Something I think we should address down the road.  I have added a todo.

Reply:
Of course, my bad 🤦